src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{
+    device_type::{BlockDevice, Device, DeviceArgument, GenericConfig, GenericDevice},
+    utils::get_virt_drive_name,
+    Hypervisor,
+};
+use agent::types::Device as AgentDevice;
+use anyhow::{anyhow, Result};
+use ini::Ini;
+use kata_sys_util::rand;
+use std::{collections::HashMap, str, sync::Arc};
+use tokio::sync::Mutex;
+/// VirtioMmio indicates block driver is virtio-mmio based
+pub const VIRTIO_MMIO: &str = "virtio-mmio";
+pub const VIRTIO_BLOCK: &str = "virtio-blk";
+pub const VFIO: &str = "vfio";
+const SYS_DEV_PREFIX: &str = "/sys/dev";
+pub const KATA_MMIO_BLK_DEV_TYPE: &str = "mmioblk";
+pub const KATA_BLK_DEV_TYPE: &str = "blk";
+type ArcBoxDevice = Arc<Mutex<Box<dyn Device>>>;
+
+// Device manager will manage the lifecycle of sandbox device
+pub struct DeviceManager {
```

Comment:
What's about the design of the `struct DeviceManager {}` and if it's for other devices such as VFIO/vsock...?

Reply:
Just refactor this part. It's able to support multiple device now